### PR TITLE
Adding transactionHash to DecentralandUtils

### PR DESCRIPTION
**Pull request by marcelomorgado**
### Issue link

https://github.com/tasitlabs/tasit/issues/173

### Types of changes

New feature (non-breaking change that adds functionality)

### CODE REVIEW

File: packages/tasit-sdk/src/helpers/DecentralandUtils.js [hunk 1]
```diff
@@ -188,6 +197,14 @@ export default class DecentralandUtils {
       ...filter,
       fromBlock,
     });
-    return logs.map(log => iface.parseLog(log));
+
+    const parsedLogs = logs.map(log => {
+      const { transactionHash } = log;
+      const parsedLog = iface.parseLog(log);
```
Comment (pcowgill) [inline packages/tasit-sdk/src/helpers/DecentralandUtils.js]: What is iface again?

File: packages/tasit-sdk/src/helpers/DecentralandUtils.js [hunk 2]
```diff
@@ -188,6 +197,14 @@ export default class DecentralandUtils {
       ...filter,
       fromBlock,
     });
-    return logs.map(log => iface.parseLog(log));
+
+    const parsedLogs = logs.map(log => {
```
Comment (pcowgill) [inline packages/tasit-sdk/src/helpers/DecentralandUtils.js]: This function is definitely low-level enough that we'll eventually want to move it to the Tasit SDK

File: packages/tasit-sdk/src/helpers/DecentralandUtils.test.js [hunk 3]
```diff
@@ -65,34 +65,51 @@ describe("DecentralandUtils", () => {
...
     expect(parcelIds).deep.equal([]);
   });
+
+  it("should expose transactionHash", async () => {
+    const sellerAssets = await getAssetsOf(sellerAddress);
+    expect(sellerAssets).to.not.be.empty;
+
+    for (let asset of sellerAssets) {
+      let { transactionHash } = asset;
+      expect(transactionHash).to.be.ok;
```
Comment (pcowgill) [inline packages/tasit-sdk/src/helpers/DecentralandUtils.test.js]: Make a TODO to assert something more specific? Or is it non-deterministic for a given test run? Maybe if we stored the hash from the action after doing a transfer, then did `getAssetsOf` and compared that the `transactionHash` from that function with the action's hash?

File: packages/tasit-sdk/src/helpers/DecentralandUtils.js (hunk 1 above)
Comment (marcelomorgado) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.js]: Its an `ethers.js` helpers class for dealing with ABI, in that case it's transforming a `{ 0, 1, 2 }` transfer event object to `{ from, to, assetId }`.

File: packages/tasit-sdk/src/helpers/DecentralandUtils.js (hunk 2 above)
Comment (marcelomorgado) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.js]: Should I extract it to an issue or create a comment is enough?

File: packages/tasit-sdk/src/helpers/DecentralandUtils.test.js (hunk 3 above)
Comment (marcelomorgado) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.test.js]: Make sense, I will do that on `Decentraland.test.js` test suite.

Comment (pcowgill) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.test.js]: Okay, sounds good. Thanks!

File: packages/tasit-sdk/src/helpers/DecentralandUtils.js (hunk 1 above)
Comment (pcowgill) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.js]: Ah ok, cool!

Comment (pcowgill) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.js]:
```suggestion
      // ethers.js helpers class for dealing with ABI
      // transforming a { 0, 1, 2 } transfer event object to { from, to, assetId }
      const parsedLog = iface.parseLog(log);
```

File: packages/tasit-sdk/src/helpers/DecentralandUtils.js (hunk 2 above)
Comment (pcowgill) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.js]: Up to you! No preference

Comment (marcelomorgado) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.js]: Added to already created issue https://github.com/tasitlabs/TasitSDK/issues/230

File: packages/tasit-sdk/src/helpers/DecentralandUtils.test.js (hunk 3 above)
Comment (marcelomorgado) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.test.js]: Done

File: packages/tasit-sdk/src/helpers/DecentralandUtils.js (hunk 1 above)
Comment (marcelomorgado) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.js]: Done.

Comment (pcowgill) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.js]: Thanks! Resolving

File: packages/tasit-sdk/src/helpers/DecentralandUtils.test.js (hunk 3 above)
Comment (pcowgill) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.test.js]: Thanks! Resolving

File: packages/tasit-sdk/src/helpers/DecentralandUtils.js (hunk 2 above)
Comment (pcowgill) [reply packages/tasit-sdk/src/helpers/DecentralandUtils.js]: Perfect. Resolving this.

File: packages/tasit-sdk/src/Decentraland.test.js [hunk 4]
```diff
@@ -317,13 +317,21 @@ describe("Decentraland", () => {
...
+          const buyerEstates = await _getEstatesOf(ephemeralAddress);
+          const buyerAssets = await getAssetsOf(ephemeralAddress);
 
-          expect(estatesBalance).to.equal(1);
-          expect(assetsBalance).to.equal(1);
+          expect(buyerEstates).to.have.lengthOf(1);
+          expect(buyerAssets).to.have.lengthOf(1);
+
+          const tx = await executeOrderAction.getTransaction();
+          const { hash: purcharseTxHash } = tx;
```
Comment (pcowgill) [inline packages/tasit-sdk/src/Decentraland.test.js]:
```suggestion
          const { hash: purchaseTxHash } = tx;
```

File: packages/tasit-sdk/src/Decentraland.test.js [hunk 5]
```diff
@@ -317,13 +317,21 @@ describe("Decentraland", () => {
...
+          const [buyerEstate] = buyerEstates;
+          const [buyerAsset] = buyerAssets;
+          const { transactionHash: estateTxHash } = buyerEstate;
+          const { transactionHash: assetTxHash } = buyerAsset;
+
+          expect(estateTxHash).to.equal(purcharseTxHash);
```
Comment (pcowgill) [inline packages/tasit-sdk/src/Decentraland.test.js]:
```suggestion
          expect(estateTxHash).to.equal(purchaseTxHash);
```

File: packages/tasit-sdk/src/Decentraland.test.js [hunk 6]
```diff
@@ -317,13 +317,21 @@ describe("Decentraland", () => {
...
+          expect(assetTxHash).to.equal(purcharseTxHash);
```
Comment (pcowgill) [inline packages/tasit-sdk/src/Decentraland.test.js]:
```suggestion
          expect(assetTxHash).to.equal(purchaseTxHash);
```

File: packages/tasit-sdk/src/Decentraland.test.js [hunk 7]
```diff
@@ -354,13 +362,21 @@ describe("Decentraland", () => {
...
-          const { length: parcelsBalance } = parcelIds;
-          const { length: assetsBalance } = assetIds;
+          const buyerParcels = await _getParcelsOf(ephemeralAddress);
+          const buyerAssets = await getAssetsOf(ephemeralAddress);
+
+          expect(buyerParcels).to.have.lengthOf(1);
+          expect(buyerAssets).to.have.lengthOf(1);
+
+          const tx = await executeOrderAction.getTransaction();
+          const { hash: purcharseTxHash } = tx;
```
Comment (pcowgill) [inline packages/tasit-sdk/src/Decentraland.test.js]:
```suggestion
          const { hash: purchaseTxHash } = tx;
```

File: packages/tasit-sdk/src/Decentraland.test.js [hunk 8]
```diff
@@ -354,13 +362,21 @@ describe("Decentraland", () => {
...
+          const [buyerParcel] = buyerParcels;
+          const [buyerAsset] = buyerAssets;
+          const { transactionHash: parcelTxHash } = buyerParcel;
+          const { transactionHash: assetTxHash } = buyerAsset;
 
-          expect(parcelsBalance).to.equal(1);
-          expect(assetsBalance).to.equal(1);
+          expect(parcelTxHash).to.equal(purcharseTxHash);
```
Comment (pcowgill) [inline packages/tasit-sdk/src/Decentraland.test.js]:
```suggestion
          expect(parcelTxHash).to.equal(purchaseTxHash);
```

File: packages/tasit-sdk/src/Decentraland.test.js [hunk 9]
```diff
@@ -354,13 +362,21 @@ describe("Decentraland", () => {
...
+          expect(assetTxHash).to.equal(purcharseTxHash);
```
Comment (pcowgill) [inline packages/tasit-sdk/src/Decentraland.test.js]:
```suggestion
          expect(assetTxHash).to.equal(purchaseTxHash);
```

**Comment by pcowgill** [comment]
@marcelomorgado I'll merge this when the tests finish running

**Pull request merged**